app/src/main/java/io/gnosis/safe/ui/transactions/details/TransactionDetailsFragment.kt

```diff
@@ -60,7 +63,13 @@ class TransactionDetailsFragment : BaseViewBindingFragment<FragmentTransactionDe
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-
+        if (requirePasscode && !passcodeShown) {
+            Navigation.findNavController(requireActivity(), R.id.nav_host).navigate(R.id.enterPasscodeFragment, Bundle().apply {
```

Comment:
findNavController() should be enough (insead of Navigation.findNavController(requireActivity(), R.id.nav_host))
Also I would move the whole nav code to a private method for a better readability (setupPasscode() or something like that)